kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/CheckpointInstance.java

```diff
@@ -61,6 +88,12 @@ public CheckpointInstance(long version) {
     public CheckpointInstance(long version, Optional<Integer> numParts) {
         this.version = version;
         this.numParts = numParts;
+        this.filePath = Optional.empty();
+        if (numParts.orElse(0) == 0) {
+            this.format = CheckpointFormat.CLASSIC;
```

Comment:
This constructor doesn't have the filepath, so there's no way for us to distinguish the two. It also doesn't matter, because this constructor is only used to instantiate a checkpoint version after all commits to be used for comparisons, so defaulting to classic won't change anything.

Reply:
When finding the last checkpoint before a version, we need a comparison `CheckpointInstance` as the first checkpoint instance after this version (for example, when getting a snapshot at version 3, we would use a comparison `CheckpointInstance` with version 4). This constructor is used to create this comparison `CheckpointInstance`, which is guaranteed to come after all checkpoints we want to consider. As a result, defaulting to classic checkpoint gives us the behavior we want:
- This `CheckpointInstance` will be greater than all checkpoints at or prior to the version to checkpoint.
- This `CheckpointInstance` will be less than or equal to all checkpoints after the version to checkpoint (since V2 checkpoints > Classic checkpoints).